src/modules/endpoint/models/CategoriesModel.js

```diff
@@ -0,0 +1,79 @@
+import normalizeUrl from 'normalize-url'
+
+/**
+ * Contains a Map [string -> CategoryModel] and some helper functions
+ */
+class CategoriesModel {
+  /**
+   * Creates a Map [url -> category] from the categories provided,
+   * whose parent attributes are first changed from id to url
+   * @param categories
+   */
+  constructor (categories = []) {
+    categories.forEach(category => {
+      if (category.id !== 0) {
+        // every category except from the root category should have a parent, so we don't have to check if it exists
+        const parentUrl = categories.find(_category => _category.id === category.parent).url
+        category.setParent(parentUrl)
+      }
+    })
```

Comment:
regarding ii) I agree,
for the rest I like it better to have the url as parent, expecially because this is only a workaround until the changes in the cms are applied, than we get the url anyway, but we can rename category.parent to category.parentUrl

Reply:
i would just keep this as it is till the changes in the cms are applied